floodviz/utils.py

```
@@ -0,0 +1,82 @@
+                    'agency' - service specific string for url construction. Cannot be used with 'site_status' key
+
+        RETURNS:
+            url - url to request for data
+    """
+    all_data = []
+    sites = None
+    url = endpoint
+    data_format = '?format=rdb'
+    if params.get('sites') is not None:
+        sites = ','.join(params['sites'])
+        if params.get('extra'):
+            url += params['extra'] + data_format
+        else:
+            url += data_format
+    # Needed
+    if params.get('site_query'):
+        url += params['site_query'] + sites
+    # Can have one or the other or both for a valid url
+    if params.get('start_date_query'):
+        url += params['start_date_query'] + params['start_date']
+    if params.get('end_date_query'):
+        url += params['end_date_query'] + params['end_date']
+    # One or the other here. Not both. 'site_status' and 'agency' keys should not be in the same params dict.
+    if params.get('site_status'):
+        url += params['site_status']
+    elif params.get('agency'):
+        url += params['agency']
+    # request url
+    try:
+        r = requests.get(url)
+    except requests.exceptions.RequestException as e:
+        print(e)
+        print('- bad webservice url -')
+    else:
+        if r.status_code is 200:
+            content = r.text.splitlines()
+            headers = None
+            for line in content:
+                # Skip commented lines
+                if line.startswith('#'):
+                    continue
+                if line.startswith('USGS'):
+                    # extraction procedure
+                    line = line.split('\t')
+                    data_point = {}
+                    # sanity check that headers has been filled
+                    if headers is not None:
+                        for idx, data in enumerate(line):
+                            data_point[headers[idx]] = data
```

This seems equivalent to:
`data_point = dict(zip(headers, line))`

It looks like this (either as-is or using `zip`) will fail if any field is missing from the line. I'm not sure if we have to take that into account, but I'd like to be sure that the parser will either always return correct data or fail* rather than returning bad data.

*Correct meaning that it can assign the right value to the right header even if a value is missing from the line, and fail meaning either throw an error or return None.